### PR TITLE
Add DeepCopy fail test

### PR DESCRIPTION
**Pull request by soltysh**
@sttts fyi, since you asked for it in https://github.com/openshift/api/pull/20#discussion_r157748678

### CODE REVIEW

File: pkg/testing/deepcopy_test.go [hunk 1]
```diff
@@ -74,6 +76,31 @@ func TestRoundTripTypesWithoutProtobuf(t *testing.T) {
...
+	builder.AddToScheme(scheme)
+	seed := rand.Int63()
+	fuzzer := fuzzer.FuzzerFor(genericfuzzer.Funcs, rand.NewSource(seed), codecs)
+	gvk := groupVersion.WithKind("BrokenType")
+	roundtrip.RoundTripSpecificKindWithoutProtobuf(t, gvk, scheme, codecs, fuzzer, nil)
+	// It's very hacky way of making sure the DeepCopy is actually invoked inside RoundTripSpecificKindWithoutProtobuf
+	// used in the other test. If for some reason this tests starts passing we need to fail b/c we're not testing
+	// the DeepCopy in the other method which we care so much about.
+	if t.Failed() {
+		os.Exit(0)
```
Comment (sttts) [inline pkg/testing/deepcopy_test.go]: doesn't this skip the other test coming after this?

Comment (soltysh) [reply pkg/testing/deepcopy_test.go]: Fixed with a temporary `testing.T`

**Pull request merged**